### ISSUE TITLE
dgesdd_ exported from shared library but not defined in stub

### ISSUE THREAD

**Issue by bktycrg** [xianyi/OpenBLAS#4592]
When building with CMake, this results in failed linking operations from consumers that expect to resolve `dgesdd_` at link time; I have verified this against `develop` and `v0.3.26`.

![image](https://github.com/OpenMathLib/OpenBLAS/assets/144853309/9cde03ae-88bf-4639-b355-8270e30f9348)

I am arriving here from "xtensor-blas" where `cxxlapack::gesdd()` is attempting to invoke this particular method, having build OpenBLAS with C_LAPACK options for shared libraries.

![image](https://github.com/OpenMathLib/OpenBLAS/assets/144853309/5ca90ec9-b993-49ab-a3f9-45b838a9d621)

I'm not sure this is a bug. In fact, I suspect it's related to Fortran symbol translation assumptions in a MSVC-driven CMake build on Windows, using the experimental build configuration thereof.

![image](https://github.com/OpenMathLib/OpenBLAS/assets/144853309/d0a5d0ac-db6c-4d47-9172-96a6e0c8522f)

But... I can't rule out the possibility that there is some macro-rewrite magic I am missing here, since `lapack.h` does define a `LAPACK_dgesdd_base` symbol, and `lapacke.h` includes a `LAPACKE_dgesdd_work` symbol (that *is* defined in the DLL exports).

![image](https://github.com/OpenMathLib/OpenBLAS/assets/144853309/4121c2d1-a58b-4dd0-86fb-98a8676d1231)

In summary:

![image](https://github.com/OpenMathLib/OpenBLAS/assets/144853309/d3a757b8-e3ec-4a27-a2a0-6105aca23ad9)

**Comment by martin-frbg** [comment]
Hmm. lapack.h should define a `LAPACK_dgesdd` based on the `LAPACK_dgesdd_base` as well (two choices depending on "hidden" size arguments for the Fortran string arguments should be visible in the C prototype or not), lapacke.h defines both LAPACKE_dgesdd and LAPACKE_dgesdd_work, and the library should definitely contain a symbol named `dgesdd_` in addition to the two LAPACKE symbols. 
The `LAPACK_` prefix on the symbol names gets bitten off by the `LAPACK_GLOBAL` macro that is defined in lapacke_mangling.h (included by lapacke.h) so you should not see nor expect `LAPACK_dgesdd_` in the wild, only `dgesdd_` itself. 
That symbol is definitely present in Linux builds (both gmake and cmake), and I see no reason why it would be left out in Windows builds. Can you check with `dumpbin /symbols /exports` or similar) what your build of OpenBLAS has ? (I assume the screenshot you included above is from the caller, not the library ?)

**Comment by martin-frbg** [comment]
Any update to this saga ? You would probably be better off building with LLVM  or oneAPI rather than MSVC (the latter can't handle all our carefully crafted assembly so shows poorer performance, also the alternatives have an actual Fortran compiler) but I believe dgesdd should be present in the generated library in any case

**Comment by bktycrg** [comment]
Managed to get consistent symbols with a cleaned and tweaked build configuration; there are other outstanding issues but I've decided to take another route for now. I will close this in the meantime.
